### DISCUSSION TITLE
Incorporate A Language Translator For (at least) CHATS!

### DISCUSSION THREAD

**Discussion by DeathStalker77** [nicotine-plus/nicotine-plus#3556] [Ideas]
### Describe the solution you'd like
I would like the CLIENT to have the ability to SET the user's PREFERRED LANGUAGE in the SETTINGS and the automatically TRANSLATE that to the PREFERRED LANGUAGE of those they communicate with.  So that copy/pasting into Google Translate is not necessary.

### Additional context
Not sure if this could be done with an API or other integration methods - I also understand that, in all likelihood, it will ONLY function between 2 N+ clients.  That is *certainly* acceptable!  Thank you!

I think CHATS are the most important, but if it IS achieved for CHATS, then maybe it could also be applied to ROOMS as well.

**Comment by mathiascode** [comment]
Seems like something that should be implemented in a plugin.

**Comment by slook** [comment]
_"Argos Translate uses [OpenNMT](https://opennmt.net/) for translations and can be used as either a Python library, command-line, or GUI application. Argos Translate supports installing language model packages which are zip archives with a ".argosmodel" extension containing the data needed for translation. [LibreTranslate](https://libretranslate.com/) is an API and web-app built on top of Argos Translate:"_

https://www.argosopentech.com/
https://github.com/argosopentech/argos-translate
https://github.com/argosopentech/LibreTranslate-py - _Python bindings to connect to a LibreTranslate API_ 

_"LibreTranslate is an Free and Open Source Machine Translation API, entirely self-hosted. Unlike other APIs, it doesn't rely on proprietary providers such as Google or Azure to perform translations. Instead, its translation engine is powered by the open source [Argos Translate](https://github.com/argosopentech/argos-translate) library:"_

https://libretranslate.com/
https://github.com/LibreTranslate/LibreTranslate
https://pypi.org/project/libretranslatepy/ - _Python bindings to connect to a LibreTranslate API_

**Comment by mathiascode** [comment]
Since this request requires either larger dependencies or external APIs, it won't be added as a core feature, but I'll convert this to a discussion in case someone is interested in creating a plugin.
